### PR TITLE
Adds organization settings links to user nav

### LINKED ISSUE #93: Multiple organization UX
Rather than listing all organizations across the top, I think we can fit them into the user dropdown on top.  For now, these will just link to the existing dashboard's organizations#show pages. 

205e organizations

### PR DESCRIPTION
**Pull request by sandersonet**
Fixes #93 

This PR requires https://github.com/aptible/auth.aptible.com/pull/119 to work properly.

For you @mixonic.

### CODE REVIEW

**Comment by sandersonet** [comment]
@mixonic aptible/auth.aptible.com#119 will cause issues by exposing organizations to unauthorized users, so I instead use a computed property on `models/user.js` that returns all unique organization through roles.  Let me know if there is a better way to do this.

File: app/models/user.js [hunk 1]
```diff
@@ -17,12 +17,35 @@ export default DS.Model.extend({
   // relationships
   token: DS.belongsTo('token', {async: true}),
   roles: DS.hasMany('role', {async:true}),
-  organizations: DS.hasMany('organization', {async:true}),
+  //organizations: DS.hasMany('organization', {async:true}),
```
Comment (mixonic) [inline app/models/user.js]: Should this line just be removed?

File: app/models/user.js [hunk 2]
```diff
@@ -17,12 +17,35 @@ export default DS.Model.extend({
...
+        };
+        organizations.push(organization);
+      }
+    });
+
+    return organizations.map(function(organization) {
+      return organization.organization;
+    });
+
+  }.property('roles.[]')
```
Comment (mixonic) [inline app/models/user.js]: I would simplify:

```javascript
  organizations: function() {
    return this.get('roles').mapBy('organization').uniq();
  }.property('roles.@each.organization')
```

Some references, I know this uses some slick trickery:

* [uniq](http://emberjs.com/api/classes/Ember.Enumerable.html#method_uniq)
* [mapBy](http://emberjs.com/api/classes/Ember.Enumerable.html#method_mapBy)
* [@each](http://emberjs.com/guides/object-model/computed-properties-and-aggregate-data/)

It should have the same effect though, and perform maybe a little better.

File: app/templates/-current-user-header.hbs [hunk 3]
```diff
@@ -6,34 +6,18 @@
...
           {{#dashboard-dropdown title=session.currentUser.name defaultPath="index"}}
 
-            {{#dashboard-dropdown-item}}
-              {{link-to "Dashboard" "index"}}
-            {{/dashboard-dropdown-item}}
-
-            {{#dashboard-dropdown-item}}
-              {{#link-to "settings"}}Settings{{/link-to}}
-            {{/dashboard-dropdown-item}}
+            {{#each organization in session.currentUser.organizations}}
```
Comment (mixonic) [inline app/templates/-current-user-header.hbs]: The block params style is preferred:

```handlebars
{{#each session.currentUser.organizations as |organization|}}
  {{dashboard-dropdown-organization-menu organization=organization}}
  {{dashboard-dropdown-divider}}
{{/each}}
```

Though not required. The `in` is basically remapped at compile time to what I wrote out here.

**Comment by mixonic** [comment]
Some failing tests and I left some comments. I an generall :+1: with the rest though!

File: app/models/user.js (hunk 2 above)
Comment (sandersonet) [inline app/models/user.js]: Thanks Matt, this is an awesome simplification.

`uniq()` doesn't seem to work though.  According to the docs:

> This only works on primitive data types, e.g. Strings, Numbers, etc.

Is there a way to get it to work with models?

Comment (mixonic) [inline app/models/user.js]: Shoot, you are right. Looks like it needs to be a bit more complex then:

```javascript
var organizations = [];
this.get('roles').forEeach(function(role){
  var organization = role.get('organization');
  if (organizations.indexOf(organization) === -1) {
    organizations.push(organization);
  }
});
return organizations;
```

**Comment by sandersonet** [comment]
cc @mixonic 

I updated the `stacks` route to include all of a user's organizations and use those to loop through and generate the `stacks` page.

File: app/templates/-current-user-header.hbs [hunk 4]
```diff
@@ -6,34 +6,18 @@
...
           {{#dashboard-dropdown title=session.currentUser.name defaultPath="index"}}
 
-            {{#dashboard-dropdown-item}}
-              {{link-to "Dashboard" "index"}}
-            {{/dashboard-dropdown-item}}
-
-            {{#dashboard-dropdown-item}}
-              {{#link-to "settings"}}Settings{{/link-to}}
-            {{/dashboard-dropdown-item}}
+            {{#each session.currentUser.organizations as |organization|}}
```
Comment (mixonic) [inline app/templates/-current-user-header.hbs]: This is where I might expect that organizations (already fetch by the route) is referenced.

**Comment by mixonic** [comment]
@sandersonet I think there are some opportunities to reduce the number of organizations queries by fetching them from a top-level route. It feels like we are implying an `organizations` route at the top level that always fetches.

But let us keep moving forward. Excellent work here!

**Pull request merged**